docs/DeveloperGuide.md

```diff
@@ -4,22 +4,194 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Design
+
+#### Architecture
+
+{Description of overall architecture}
+
+#### Ui Component
+
+The structure of the Ui Component is illustrated in the following diagram:
+
+![UiComponentDiagram](images/UiComponent.png)
+
+API : `Ui.java`
+
+1. The `Ui` class is used to communicate with users.
+2. The `Ui` class depends on `Scanner` to get inputs from users.
+3. The `Main` class uses `Ui#getNextLine()` to receive `commands` from users.
+4. And `FlashcardFactory` and `GroupFactory` classes uses methods in `Ui` class to complete 
+the execution of various commands.
+
+Given below is the Sequence Diagram for interactions between the `Ui` and `FlashcardFactory` components to complete the
+execution of `deleteCommand`.
+
+![UiSequenceDiagram](images/UiSequenceDiagram.png)
```

Comment:
In the sequence diagram, there is a FlashcardList class that interacts with UI class. However, in the class diagram, there is no FlashcardList class. Should you add it in the class diagram?